File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/registered_model.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class RegisteredModel(object):
    """
    Details of the registered model
    """

    #: A constant which can be used with the deployment_job_state property of a RegisteredModel.
    #: This constant has a value of "DEPLOYMENT_JOB_CONNECTION_STATE_UNSPECIFIED"
    DEPLOYMENT_JOB_STATE_DEPLOYMENT_JOB_CONNECTION_STATE_UNSPECIFIED = "DEPLOYMENT_JOB_CONNECTION_STATE_UNSPECIFIED"

    #: A constant which can be used with the deployment_job_state property of a RegisteredModel.
    #: This constant has a value of "NOT_SET_UP"
    DEPLOYMENT_JOB_STATE_NOT_SET_UP = "NOT_SET_UP"

    #: A constant which can be used with the deployment_job_state property of a RegisteredModel.
    #: This constant has a value of "CONNECTED"
    DEPLOYMENT_JOB_STATE_CONNECTED = "CONNECTED"

    #: A constant which can be used with the deployment_job_state property of a RegisteredModel.
    #: This constant has a value of "NOT_FOUND"
    DEPLOYMENT_JOB_STATE_NOT_FOUND = "NOT_FOUND"

    #: A constant which can be used with the deployment_job_state property of a RegisteredModel.
    #: This constant has a value of "REQUIRED_PARAMETERS_CHANGED"
    DEPLOYMENT_JOB_STATE_REQUIRED_PARAMETERS_CHANGED = "REQUIRED_PARAMETERS_CHANGED"

    def __init__(self, **kwargs):
        """
        Initializes a new RegisteredModel object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param name:
            The value to assign to the name property of this RegisteredModel.
        :type name: str

        :param creation_timestamp:
            The value to assign to the creation_timestamp property of this RegisteredModel.
        :type creation_timestamp: int

        :param last_updated_timestamp:
            The value to assign to the last_updated_timestamp property of this RegisteredModel.
        :type last_updated_timestamp: int

        :param description:
            The value to assign to the description property of this RegisteredModel.
        :type description: str

        :param latest_versions:
            The value to assign to the latest_versions property of this RegisteredModel.
        :type latest_versions: list[oci.aidataplatform_dp.models.ModelVersion]

        :param aliases:
            The value to assign to the aliases property of this RegisteredModel.
        :type aliases: list[oci.aidataplatform_dp.models.RegisteredModelAlias]

        :param deployment_job_id:
            The value to assign to the deployment_job_id property of this RegisteredModel.
        :type deployment_job_id: str

        :param deployment_job_state:
            The value to assign to the deployment_job_state property of this RegisteredModel.
            Allowed values for this property are: "DEPLOYMENT_JOB_CONNECTION_STATE_UNSPECIFIED", "NOT_SET_UP", "CONNECTED", "NOT_FOUND", "REQUIRED_PARAMETERS_CHANGED", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type deployment_job_state: str

        :param tags:
            The value to assign to the tags property of this RegisteredModel.
        :type tags: list[oci.aidataplatform_dp.models.RegisteredModelTag]

        """
        self.swagger_types = {
            'name': 'str',
            'creation_timestamp': 'int',
            'last_updated_timestamp': 'int',
            'description': 'str',
            'latest_versions': 'list[ModelVersion]',
            'aliases': 'list[RegisteredModelAlias]',
            'deployment_job_id': 'str',
            'deployment_job_state': 'str',
            'tags': 'list[RegisteredModelTag]'
        }

        self.attribute_map = {
            'name': 'name',
            'creation_timestamp': 'creation_timestamp',
            'last_updated_timestamp': 'last_updated_timestamp',
            'description': 'description',
            'latest_versions': 'latest_versions',
            'aliases': 'aliases',
            'deployment_job_id': 'deployment_job_id',
            'deployment_job_state': 'deployment_job_state',
            'tags': 'tags'
        }

        self._name = None
        self._creation_timestamp = None
        self._last_updated_timestamp = None
        self._description = None
        self._latest_versions = None
        self._aliases = None
        self._deployment_job_id = None
        self._deployment_job_state = None
        self._tags = None

    @property
    def name(self):
        """
        Gets the name of this RegisteredModel.
        Unique name for the model.


        :return: The name of this RegisteredModel.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this RegisteredModel.
        Unique name for the model.


        :param name: The name of this RegisteredModel.
        :type: str
        """
        self._name = name

    @property
    def creation_timestamp(self):
        """
        Gets the creation_timestamp of this RegisteredModel.
        Timestamp in milliseconds when the model was created.


        :return: The creation_timestamp of this RegisteredModel.
        :rtype: int
        """
        return self._creation_timestamp

    @creation_timestamp.setter
    def creation_timestamp(self, creation_timestamp):
        """
        Sets the creation_timestamp of this RegisteredModel.
        Timestamp in milliseconds when the model was created.


        :param creation_timestamp: The creation_timestamp of this RegisteredModel.
        :type: int
        """
        self._creation_timestamp = creation_timestamp

    @property
    def last_updated_timestamp(self):
        """
        Gets the last_updated_timestamp of this RegisteredModel.
        Timestamp in milliseconds when metadata for the model was last updated.


        :return: The last_updated_timestamp of this RegisteredModel.
        :rtype: int
        """
        return self._last_updated_timestamp

    @last_updated_timestamp.setter
    def last_updated_timestamp(self, last_updated_timestamp):
        """
        Sets the last_updated_timestamp of this RegisteredModel.
        Timestamp in milliseconds when metadata for the model was last updated.


        :param last_updated_timestamp: The last_updated_timestamp of this RegisteredModel.
        :type: int
        """
        self._last_updated_timestamp = last_updated_timestamp

    @property
    def description(self):
        """
        Gets the description of this RegisteredModel.
        Description of the registered model.


        :return: The description of this RegisteredModel.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this RegisteredModel.
        Description of the registered model.


        :param description: The description of this RegisteredModel.
        :type: str
        """
        self._description = description

    @property
    def latest_versions(self):
        """
        Gets the latest_versions of this RegisteredModel.
        Collection of latest model versions for each stage. Only contains models with current READY status.


        :return: The latest_versions of this RegisteredModel.
        :rtype: list[oci.aidataplatform_dp.models.ModelVersion]
        """
        return self._latest_versions

    @latest_versions.setter
    def latest_versions(self, latest_versions):
        """
        Sets the latest_versions of this RegisteredModel.
        Collection of latest model versions for each stage. Only contains models with current READY status.


        :param latest_versions: The latest_versions of this RegisteredModel.
        :type: list[oci.aidataplatform_dp.models.ModelVersion]
        """
        self._latest_versions = latest_versions

    @property
    def aliases(self):
        """
        Gets the aliases of this RegisteredModel.
        Aliases pointing to model versions associated with this registered_model.


        :return: The aliases of this RegisteredModel.
        :rtype: list[oci.aidataplatform_dp.models.RegisteredModelAlias]
        """
        return self._aliases

    @aliases.setter
    def aliases(self, aliases):
        """
        Sets the aliases of this RegisteredModel.
        Aliases pointing to model versions associated with this registered_model.


        :param aliases: The aliases of this RegisteredModel.
        :type: list[oci.aidataplatform_dp.models.RegisteredModelAlias]
        """
        self._aliases = aliases

    @property
    def deployment_job_id(self):
        """
        Gets the deployment_job_id of this RegisteredModel.
        Deployment job ID.


        :return: The deployment_job_id of this RegisteredModel.
        :rtype: str
        """
        return self._deployment_job_id

    @deployment_job_id.setter
    def deployment_job_id(self, deployment_job_id):
        """
        Sets the deployment_job_id of this RegisteredModel.
        Deployment job ID.


        :param deployment_job_id: The deployment_job_id of this RegisteredModel.
        :type: str
        """
        self._deployment_job_id = deployment_job_id

    @property
    def deployment_job_state(self):
        """
        Gets the deployment_job_state of this RegisteredModel.
        Job state.

        Allowed values for this property are: "DEPLOYMENT_JOB_CONNECTION_STATE_UNSPECIFIED", "NOT_SET_UP", "CONNECTED", "NOT_FOUND", "REQUIRED_PARAMETERS_CHANGED", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The deployment_job_state of this RegisteredModel.
        :rtype: str
        """
        return self._deployment_job_state

    @deployment_job_state.setter
    def deployment_job_state(self, deployment_job_state):
        """
        Sets the deployment_job_state of this RegisteredModel.
        Job state.


        :param deployment_job_state: The deployment_job_state of this RegisteredModel.
        :type: str
        """
        allowed_values = ["DEPLOYMENT_JOB_CONNECTION_STATE_UNSPECIFIED", "NOT_SET_UP", "CONNECTED", "NOT_FOUND", "REQUIRED_PARAMETERS_CHANGED"]
        if not value_allowed_none_or_none_sentinel(deployment_job_state, allowed_values):
            deployment_job_state = 'UNKNOWN_ENUM_VALUE'
        self._deployment_job_state = deployment_job_state

    @property
    def tags(self):
        """
        Gets the tags of this RegisteredModel.
        Tags for the registered model.


        :return: The tags of this RegisteredModel.
        :rtype: list[oci.aidataplatform_dp.models.RegisteredModelTag]
        """
        return self._tags

    @tags.setter
    def tags(self, tags):
        """
        Sets the tags of this RegisteredModel.
        Tags for the registered model.


        :param tags: The tags of this RegisteredModel.
        :type: list[oci.aidataplatform_dp.models.RegisteredModelTag]
        """
        self._tags = tags

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/RegisteredModel.java
package com.oracle.aidataplatform.dp.model;


/**
 * Details of the registered model
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=RegisteredModel.Builder.class)

public final class RegisteredModel  {
    @Deprecated
    @java.beans.ConstructorProperties({"name", "creationTimestamp", "lastUpdatedTimestamp", "description", "latestVersions", "aliases", "deploymentJobId", "deploymentJobState", "tags"})
    public RegisteredModel(String name, Long creationTimestamp, Long lastUpdatedTimestamp, String description, java.util.List<ModelVersion> latestVersions, java.util.List<RegisteredModelAlias> aliases, String deploymentJobId, DeploymentJobState deploymentJobState, java.util.List<RegisteredModelTag> tags) {
        super();
        this.name = name;
        this.creationTimestamp = creationTimestamp;
        this.lastUpdatedTimestamp = lastUpdatedTimestamp;
        this.description = description;
        this.latestVersions = latestVersions;
        this.aliases = aliases;
        this.deploymentJobId = deploymentJobId;
        this.deploymentJobState = deploymentJobState;
        this.tags = tags;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Unique name for the model.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("name")
        private String name;

                /**
         * Unique name for the model.
         * @param name the value to set
         * @return this builder
         **/
        

        public Builder name(String name) {
        this.name = name;
        return this;
        }
            /**
     * Timestamp in milliseconds when the model was created.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("creation_timestamp")
        private Long creationTimestamp;

                /**
         * Timestamp in milliseconds when the model was created.
         * @param creationTimestamp the value to set
         * @return this builder
         **/
        

        public Builder creationTimestamp(Long creationTimestamp) {
        this.creationTimestamp = creationTimestamp;
        return this;
        }
            /**
     * Timestamp in milliseconds when metadata for the model was last updated.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("last_updated_timestamp")
        private Long lastUpdatedTimestamp;

                /**
         * Timestamp in milliseconds when metadata for the model was last updated.
         * @param lastUpdatedTimestamp the value to set
         * @return this builder
         **/
        

        public Builder lastUpdatedTimestamp(Long lastUpdatedTimestamp) {
        this.lastUpdatedTimestamp = lastUpdatedTimestamp;
        return this;
        }
            /**
     * Description of the registered model.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("description")
        private String description;

                /**
         * Description of the registered model.
         * @param description the value to set
         * @return this builder
         **/
        

        public Builder description(String description) {
        this.description = description;
        return this;
        }
            /**
     * Collection of latest model versions for each stage. Only contains models with current READY status.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("latest_versions")
        private java.util.List<ModelVersion> latestVersions;

                /**
         * Collection of latest model versions for each stage. Only contains models with current READY status.
         * @param latestVersions the value to set
         * @return this builder
         **/
        

        public Builder latestVersions(java.util.List<ModelVersion> latestVersions) {
        this.latestVersions = latestVersions;
        return this;
        }
            /**
     * Aliases pointing to model versions associated with this registered_model.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("aliases")
        private java.util.List<RegisteredModelAlias> aliases;

                /**
         * Aliases pointing to model versions associated with this registered_model.
         * @param aliases the value to set
         * @return this builder
         **/
        

        public Builder aliases(java.util.List<RegisteredModelAlias> aliases) {
        this.aliases = aliases;
        return this;
        }
            /**
     * Deployment job ID.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("deployment_job_id")
        private String deploymentJobId;

                /**
         * Deployment job ID.
         * @param deploymentJobId the value to set
         * @return this builder
         **/
        

        public Builder deploymentJobId(String deploymentJobId) {
        this.deploymentJobId = deploymentJobId;
        return this;
        }
            /**
     * Job state.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("deployment_job_state")
        private DeploymentJobState deploymentJobState;

                /**
         * Job state.
         * @param deploymentJobState the value to set
         * @return this builder
         **/
        

        public Builder deploymentJobState(DeploymentJobState deploymentJobState) {
        this.deploymentJobState = deploymentJobState;
        return this;
        }
            /**
     * Tags for the registered model.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("tags")
        private java.util.List<RegisteredModelTag> tags;

                /**
         * Tags for the registered model.
         * @param tags the value to set
         * @return this builder
         **/
        

        public Builder tags(java.util.List<RegisteredModelTag> tags) {
        this.tags = tags;
        return this;
        }


        public RegisteredModel build() {
            RegisteredModel model = new RegisteredModel(this.name
                    , this.creationTimestamp
                    , this.lastUpdatedTimestamp
                    , this.description
                    , this.latestVersions
                    , this.aliases
                    , this.deploymentJobId
                    , this.deploymentJobState
                    , this.tags);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(RegisteredModel model) {
                this.name(model.getName());
    this.creationTimestamp(model.getCreationTimestamp());
    this.lastUpdatedTimestamp(model.getLastUpdatedTimestamp());
    this.description(model.getDescription());
    this.latestVersions(model.getLatestVersions());
    this.aliases(model.getAliases());
    this.deploymentJobId(model.getDeploymentJobId());
    this.deploymentJobState(model.getDeploymentJobState());
    this.tags(model.getTags());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Unique name for the model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * Unique name for the model.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * Timestamp in milliseconds when the model was created.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("creation_timestamp")
    private final Long creationTimestamp;

        /**
     * Timestamp in milliseconds when the model was created.
     * @return the value
     **/
    
    public Long getCreationTimestamp() {
        return creationTimestamp;
    }


        /**
     * Timestamp in milliseconds when metadata for the model was last updated.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("last_updated_timestamp")
    private final Long lastUpdatedTimestamp;

        /**
     * Timestamp in milliseconds when metadata for the model was last updated.
     * @return the value
     **/
    
    public Long getLastUpdatedTimestamp() {
        return lastUpdatedTimestamp;
    }


        /**
     * Description of the registered model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * Description of the registered model.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * Collection of latest model versions for each stage. Only contains models with current READY status.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("latest_versions")
    private final java.util.List<ModelVersion> latestVersions;

        /**
     * Collection of latest model versions for each stage. Only contains models with current READY status.
     * @return the value
     **/
    
    public java.util.List<ModelVersion> getLatestVersions() {
        return latestVersions;
    }


        /**
     * Aliases pointing to model versions associated with this registered_model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("aliases")
    private final java.util.List<RegisteredModelAlias> aliases;

        /**
     * Aliases pointing to model versions associated with this registered_model.
     * @return the value
     **/
    
    public java.util.List<RegisteredModelAlias> getAliases() {
        return aliases;
    }


        /**
     * Deployment job ID.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("deployment_job_id")
    private final String deploymentJobId;

        /**
     * Deployment job ID.
     * @return the value
     **/
    
    public String getDeploymentJobId() {
        return deploymentJobId;
    }

    
        /**
     * Job state.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("deployment_job_state")
    private final DeploymentJobState deploymentJobState;

        /**
     * Job state.
     * @return the value
     **/
    
    public DeploymentJobState getDeploymentJobState() {
        return deploymentJobState;
    }


        /**
     * Tags for the registered model.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tags")
    private final java.util.List<RegisteredModelTag> tags;

        /**
     * Tags for the registered model.
     * @return the value
     **/
    
    public java.util.List<RegisteredModelTag> getTags() {
        return tags;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("RegisteredModel(");
        sb.append("name=").append(String.valueOf(this.name));
        sb.append(", creationTimestamp=").append(String.valueOf(this.creationTimestamp));
        sb.append(", lastUpdatedTimestamp=").append(String.valueOf(this.lastUpdatedTimestamp));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", latestVersions=").append(String.valueOf(this.latestVersions));
        sb.append(", aliases=").append(String.valueOf(this.aliases));
        sb.append(", deploymentJobId=").append(String.valueOf(this.deploymentJobId));
        sb.append(", deploymentJobState=").append(String.valueOf(this.deploymentJobState));
        sb.append(", tags=").append(String.valueOf(this.tags));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof RegisteredModel)) {
            return false;
        }

        RegisteredModel other = (RegisteredModel) o;
        return java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.creationTimestamp, other.creationTimestamp) &&
            java.util.Objects.equals(this.lastUpdatedTimestamp, other.lastUpdatedTimestamp) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.latestVersions, other.latestVersions) &&
            java.util.Objects.equals(this.aliases, other.aliases) &&
            java.util.Objects.equals(this.deploymentJobId, other.deploymentJobId) &&
            java.util.Objects.equals(this.deploymentJobState, other.deploymentJobState) &&
            java.util.Objects.equals(this.tags, other.tags);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.creationTimestamp == null ? 43 : this.creationTimestamp.hashCode());
        result = (result * PRIME) + (this.lastUpdatedTimestamp == null ? 43 : this.lastUpdatedTimestamp.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.latestVersions == null ? 43 : this.latestVersions.hashCode());
        result = (result * PRIME) + (this.aliases == null ? 43 : this.aliases.hashCode());
        result = (result * PRIME) + (this.deploymentJobId == null ? 43 : this.deploymentJobId.hashCode());
        result = (result * PRIME) + (this.deploymentJobState == null ? 43 : this.deploymentJobState.hashCode());
        result = (result * PRIME) + (this.tags == null ? 43 : this.tags.hashCode());
        return result;
    }


}
